### PR TITLE
[react-native] Minor UIManager update, and removal of dead types

### PR DESCRIPTION
**Pull request by acoates-ms**
Over time various types have been deprecated from react-native, and never removed from the types. -- Most of this is removal of types that are no longer part of react-native.

I also added a few deprecation attributes to types that will be removed soon, and throw runtime deprecation warnings if used.

Also added the getViewManagerConfig function to UIManager, since that was missing.

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/facebook/react-native
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Review by Esemesek — APPROVED** [review]
lgtmt

**Review by Titozzz — APPROVED** [review]

File: types/react-native/index.d.ts
Comment (TheSavior) [inline types/react-native/index.d.ts]: I think this might be optional, but it is typed as `Object` in the RN flow type, so...this is at least better than that...

I'm fine with this if you are.

**Review by elicwhite — APPROVED** [review]

File: types/react-native/index.d.ts
Comment (alloy) [inline types/react-native/index.d.ts]: @TheSavior :wave:

**Review by alloy — APPROVED** [review]
Great stuff 👌 

Looks like this is breaking the current styled-components typings, but in reality it should probably be updated to no longer use WebView from RN but from https://github.com/react-native-community/react-native-webview instead, which ships with an ambient declaration file.

**Comment by acoates-ms** [comment]
> Great stuff 👌
> 
> Looks like this is breaking the current styled-components typings, but in reality it should probably be updated to no longer use WebView from RN but from https://github.com/react-native-community/react-native-webview instead, which ships with an ambient declaration file.

Do you know if that break will block the PR?  I'm not that familiar with making changes on DefinitelyTyped.  - It does seem like the issue is on styled-component at this point though.

**Comment by alloy** [comment]
With #35239 & #35274 there already is precedent for removing deprecated native bits from the styled-components typings. As such, I feel it would be fair to include a change in this PR that would remove `WebView` coverage from the current styled-components typings.

/cc @JKillian @Igorbek

**Comment by orta** [comment]
I've pushed a commit which I think should fix CI, removing the webview from styled components - should unblock this PR

**Comment by alloy** [comment]
@orta Yup, that’s it—thanks!

**Comment by alloy** [comment]
I actually wanted to wait to hear from the maintainers of the styled-components typings before merging, but I should have said so explicitly. My bad.

**Pull request merged**
